xarray/tests/test_cftimeindex_resample.py

```diff
@@ -0,0 +1,134 @@
+from __future__ import absolute_import
+
+import itertools
+import pytest
+import datetime as dt
+
+import numpy as np
+import pandas as pd
+import xarray as xr
+from xarray.tests import assert_array_equal, assert_identical
+
+
+@pytest.fixture()
+def pd_index():
+    return pd.date_range('2000-01-01', periods=30, freq='MS', tz='UTC')
+
+
+@pytest.fixture()
+def xr_index():
+    return xr.cftime_range('2000-01-01', periods=30, freq='MS', tz='UTC')
+
+
+@pytest.fixture()
+def daily_pd_index():
+    return pd.date_range('2000-01-01', periods=900, freq='D', tz='UTC')
+
+
+@pytest.fixture()
+def daily_xr_index():
+    return xr.cftime_range('2000-01-01', periods=900, freq='D', tz='UTC')
+
+
+@pytest.fixture()
+def base_pd_index():
+    return pd.date_range('2000-01-01', periods=30, freq='D', tz='UTC')
+
+
+@pytest.fixture()
+def base_xr_index():
+    return xr.cftime_range('2000-01-01', periods=30, freq='D', tz='UTC')
+
+
+def da(index):
+    return xr.DataArray(np.arange(100., 100.+index.size),
+                        coords=[index], dims=['time'])
+
+
+def series(index):
+    return pd.Series(np.arange(100., 100.+index.size), index=index)
+
+
+@ pytest.mark.parametrize(('closed', 'label', 'freq'),
+                          list(itertools.product(
+                              ['left', 'right'],
+                              ['left', 'right'],
+                              ['2MS', '2M', '3MS', '3M', '7MS', '7M'])))
```

Comment:
Brief note on the test approach - overall looks great - on these items, you can put each set of params in its own `parametrize` decorator, and avoid making the products yourself

(no need to change this time though)

Reply:
Thanks for the comment, I made the change, much better than the torturous code I was writing before!